hmt_escrow/job.py

```diff
@@ -0,0 +1,1021 @@
+#!/usr/bin/env python3
+import os
+import logging
+
+from decimal import *
+from enum import Enum
+from typing import Dict, List, Tuple, Optional
+
+from web3 import Web3
+from web3.contract import Contract
+from eth_keys import keys
+from eth_utils import decode_hex
+
+from eth_bridge import get_hmtoken, get_contract_interface, get_escrow, get_factory, deploy_factory, get_w3, handle_transaction
+from storage import download, upload
+from basemodels import Manifest
+
+GAS_LIMIT = int(os.getenv("GAS_LIMIT", 4712388))
+
+LOG = logging.getLogger("hmt_escrow.job")
+Status = Enum('Status', 'Launched Pending Partial Paid Complete Cancelled')
+
+
+class Job:
+    """A class used to represent a given Job launched on the HUMAN network.
+    A Job  can be created from a manifest or by accessing an existing escrow contract 
+    from the Ethereum network. The manifest has to follow the Manifest model 
+    specification at https://github.com/hCaptcha/hmt-basemodels.
+
+    A typical Job goes through the following stages:
+    Launch: deploy an escrow contract to the network.
+    Setup: store relevant attributes in the contract state.
+    Pay: pay all job participatants in HMT when all the Job's tasks have been completed.
+
+    Attributes:
+        serialized_manifest (Dict[str, Any]): a dict representation of the Manifest model.
+        factory_contract (Contract): the factory contract used to create Job's escrow contract.
+        job_contract (Contract): the escrow contract of the Job.
+        gas_payer (str): an ethereum address paying for the gas costs.
+        gas_payer_priv (str): the private key of the gas_payer.
+        amount (Decimal): an amount to be stored in the escrow contract.
+        manifest_url (str): the location of the serialized manifest in IPFS.
+        manifest_hash (str): SHA-1 hashed version of the serialized manifest.
+
+    """
+
+    def __init__(self,
+                 credentials: Dict[str, str],
+                 escrow_manifest: Manifest = None,
+                 factory_addr: str = None,
+                 escrow_addr: str = None):
```

Comment:
This simple api is way cleaner and easy to understand.  Really like how you can point it an existing factory/ address or have it create a new one that allows you to operate off of that.